### PR TITLE
[Net] Refactor RPCs, remove RSETs

### PR DESCRIPTION
**Pull request by Faless**
In this PR:
- Removed rset
- rpc_config can now optionally configure transfer mode
  (reliable/unreliable/ordered) and channel (channels are not actually
  implemented yet.)
- Refactor how the RPC id is computed to minimize the logic in Node and
  scripts that now only needs a single `get_rpc_methods` function.

Script languages will need update to use the new features (for GDScript, I was thinking of having the rpc keywords accept 2 optional parameters for transfer_mode and channel, or, just have a single `@rpc` annotation, that takes up to 3 arguments, including the rpc_mode, i.e. master/puppet/remote/etc).

### CODE REVIEW

**Comment by reduz** [comment]
This looks good to me, @vnen should give ok. Someone (either of you) needs to implement the RPC callbables, so I guess you will need to agree on that.

File: core/io/multiplayer_api.h [hunk 1]
```diff
@@ -37,6 +37,35 @@
...
 
+public:
+	enum RPCMode {
+		RPC_MODE_DISABLED, // No rpc for this method, calls to this will be blocked (default)
+		RPC_MODE_REMOTE, // Using rpc() on it will call method / set property in all remote peers
+		RPC_MODE_MASTER, // Using rpc() on it will call method on wherever the master is, be it local or remote
+		RPC_MODE_PUPPET, // Using rpc() on it will call method for all puppets
+		RPC_MODE_REMOTESYNC, // Using rpc() on it will call method / set property in all remote peers and locally
+		RPC_MODE_MASTERSYNC, // Using rpc() on it will call method / set property in the master peer and locally
+		RPC_MODE_PUPPETSYNC, // Using rpc() on it will call method / set property in all puppets peers and locally
```
Comment (Calinou) [inline core/io/multiplayer_api.h:48]: Since `rset` has been removed, should the comments be updated to remove the references to "set property"?

**Review by vnen — APPROVED** [review]
For the script part it looks okay.

**Comment by vnen** [comment]
We have to see how the annotation is gonna work. I think having only `@rpc` is interesting.

We need to work on the custom callable too so the new calling syntax work.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Calinou** [comment]
For anyone curious about the removal of RSETs, see [this article](https://godotengine.org/article/multiplayer-changes-godot-4-0-report-1) for rationale.

**Pull request merged**